### PR TITLE
Allow to use nav with tab JS

### PR DESCRIPTION
**Pull request by Johann-S**
This PR will allow user to use ``nav`` with Tab JS

Include : 
- [x] Unit test
- [x] Visual exemple

Related to : https://github.com/twbs/bootstrap/issues/21223

### CODE REVIEW

File: js/tests/visual/tab.html [hunk 1]
```diff
@@ -163,6 +163,45 @@
           <p>Trust fund seitan letterpress, keytar raw denim keffiyeh etsy art party before they sold out master cleanse gluten-free squid scenester freegan cosby sweater. Fanny pack portland seitan DIY, art party locavore wolf cliche high life echo park Austin. Cred vinyl keffiyeh DIY salvia PBR, banh mi before they sold out farm-to-table VHS viral locavore cosby sweater. Lomo wolf viral, mustache readymade thundercats keffiyeh craft beer marfa ethical. Wolf salvia freegan, sartorial keffiyeh echo park vegan.</p>
         </div>
       </div>
+
+      <h4>Tabs with nav (with fade)</h4>
+      <nav class="nav nav-pills">
+        <div class="nav-item">
+          <a class="nav-link active" data-toggle="tab" href="#home5">Home</a>
+        </div>
```
Comment (mdo) [inline js/tests/visual/tab.html]: Is it possible for `.nav-item` to be on the anchor? Example: `<a class="nav-item nav-link active">...</a>`?

Comment (Johann-S) [inline js/tests/visual/tab.html]: Yes that's possible 👍 I'll change that thank you

File: js/tests/unit/tab.js [hunk 2]
```diff
@@ -92,6 +92,24 @@ $(function () {
     assert.strictEqual($('#qunit-fixture').find('.active').attr('id'), 'home')
   })
 
+  QUnit.test('should activate element by tab id in nav list', function (assert) {
+    assert.expect(2)
+    var tabsHTML = [
+      '<nav class="nav">',
+      '<div><a href="#home">Home</a></div>',
+      '<div><a href="#profile">Profile</a></div>',
```
Comment (mdo) [inline js/tests/unit/tab.js]: Should these lines still have `<div>`s, or can we safely remove them since we removed them from the visual example?

Comment (Johann-S) [inline js/tests/unit/tab.js]: I changed that thank you @mdo 👍

**Pull request merged**